libs/feature/record/src/lib/map-view/map-view.component.html

```diff
@@ -31,8 +31,46 @@
       >
         <ng-icon name="matClose" class="align-middle text-sm"></ng-icon>
       </button>
+      <gn-ui-button
+        type="primary"
+        (buttonClick)="resetSelection()"
+        extraClass="rounded absolute right-[0.5em] p-2 text-xs"
+      >
+        <ng-icon name="matClose" class="align-middle text-sm"></ng-icon>
+      </gn-ui-button>
       <gn-ui-feature-detail [feature]="selection"></gn-ui-feature-detail>
     </div>
+
+    <div
+      class="top-[1em] p-3 bg-white absolute overflow-y-auto overflow-x-hidden max-h-72 w-56"
+      [ngClass]="{ 'right-[1em]': !selection, 'right-[16em]': selection }"
+      [hidden]="!showLegend || !legendExists"
+    >
+      <div class="flex justify-between items-center mb-2">
+        <div class="text-primary font-bold">Legend</div>
+        <gn-ui-button
+          type="primary"
+          (buttonClick)="toggleLegend()"
+          extraClass="rounded p-2 text-xs"
```

Comment:
You probably don't need the `rounded p-2` classes here; take a look at this component for a way to have a nice and square close button:
https://github.com/geonetwork/geonetwork-ui/blob/da5f4364c0bbfb194e6a7f8265fac392c0ad0570/libs/ui/layout/src/lib/sortable-list/sortable-list.component.html#L23-L34

Reply:
@jahow Does the close button look better now?

![image](https://github.com/user-attachments/assets/19a12119-00f4-4e48-adce-af30d37ac7bf)